jwst/cube_build/ifu_cube.py

```diff
@@ -1306,13 +1309,47 @@ def setup_ifucube_wcs(self):
             log.debug(f'Working on data from {this_a}, {this_b}')
             n = len(self.master_table.FileMap[self.instrument][this_a][this_b])
             log.debug('number of files %d', n)
+            
+            # find the median center declination if we have an offset file
+            if self.offsets is not None:
+                decs = []
+                for k in range(n):
+                    input_file = self.master_table.FileMap[self.instrument][this_a][this_b][k]
+                    input_model = datamodels.open(input_file)
+                    spatial_box = input_model.meta.wcsinfo.s_region
+                    s = spatial_box.split(' ')
+                    cb1 = float(s[4])
+                    cb2 = float(s[6])
+                    cb3 = float(s[8])
+                    cb4 = float(s[10])
+                    m = (cb1 + cb2 + cb3 + cb4)/4
+                    decs.append(m)
+
+                self.median_dec = np.nanmedian(decs)
+                # fold in the median_dec information into ra offset
+                noffsets = len(self.offsets['raoffset'])
+                # convert ra and dec offsets to degrees and adjust ra offset for cos(dec) 
+                for im in range(noffsets):
+                    self.offsets['raoffset'][im] = (self.offsets['raoffset'][im]/3600.0)/np.cos(self.median_dec*deg2rad)
+                    self.offsets['decoffset'][im] = self.offsets['decoffset'][im]/3600.0
```

Comment:
Did you look into using SkyCoord to do the sky math instead?  This median dec handling is probably fine in most cases, but I always worry about edge cases with spherical coordinates.

Reply:
I did but we wanted the cos(dec) to be the same for all the time. We did not want that to vary.

Reply:
Sorry, I wasn't clear here.  Using one declination reference point is fine, my concern is that sky math has a lot of edge cases and gotchas which using something like SkyCoord naturally works around.  Doing the offset calculations by hand seems easier than setting up SkyCoord objects, but it might be more error prone.

Right now, I can't think of a counterexample where taking a simple median over declination values would be a problem, but it did occur to me over the weekend that later, when you're adding the offset to the RA and Dec, there is no wraparound check.  What happens if the declination is near 90 and the offset pushes it over?  What if the RA is near 0 or 360?  We can add checks for those cases everywhere the addition happens, or we could refactor with SkyCoord and let it do the math.

Reply:
That's not quite what I was suggesting.  What I meant by using SkyCoords was to make the corner values that eventually use the offsets SkyCoord objects, and then use something like [SkyCoord.directional_offset_by](https://docs.astropy.org/en/latest/api/astropy.coordinates.SkyCoord.html#astropy.coordinates.SkyCoord.directional_offset_by) to compute the new RA/Dec position from the input offset in arcsec.  That way it will do the cosine calculation and any necessary value wrapping under the hood.

Reply:
Actually, I think this is the function I was looking for:
[spherical_offsets_by](https://docs.astropy.org/en/latest/api/astropy.coordinates.SkyCoord.html#astropy.coordinates.SkyCoord.spherical_offsets_by)